raiden/raiden_service.py

```diff
@@ -635,6 +651,214 @@ def settle(self, token_address, partner_address):
+                partner=pex(partner_address)
+            )
+
+    def _retry_connect(self):
+        """Will be called when new channels in the token network are detected.
+        If the minimum number of channels was not yet established, it will try
+        to open new channels.
+
+        If the connection manager has no funds, this is a noop.
+        """
+        # not initialized
+        if self.funds <= 0:
+            return
+        with self.lock:
+            if self.funds_remaining <= 0:
+                return
+            if len(self.open_channels) >= self.initial_channel_target:
+                return
+            for partner in self.find_new_partners(self.initial_channel_target - len(self.open_channels)):
+                self.raiden.api.open(
+                    self.token_address.encode('hex'),
+                    partner.encode('hex')
+                )
+                self.raiden.api.deposit(
+                    self.token_address.encode('hex'),
+                    partner.encode('hex'),
+                    self.initial_funding_per_partner
+                )
+
+    def find_new_partners(self, number):
+        """Search the token network for potential channel partners.
+        Args:
+            number (int): number of partners to return
+        """
+        known = set([c.partner_address for c in self.open_channels])
+        known = known.union({self.__class__.BOOTSTRAP_ADDR_BIN})
+        known = known.union({self.raiden.address})
+        available = set(self.manager.channelgraph.graph.nodes()) - known
+
+        available = self._select_best_partners(available)
+        return available[:number]
+
+    def _select_best_partners(self, partners):
+        # FIXME: use a proper selection strategy
+        return list(partners)
+
+    @property
+    def initial_funding_per_partner(self):
+        """The calculated funding per partner depending on configuration and
+        overall funding of the ConnectionManager.
+        """
+        return int(
+                self.funds * (1 - self.joinable_funds_target)
+                / self.initial_channel_target
+                )
+
+    @property
+    def wants_more_channels(self):
+        """True, if funds available and the `initial_channel_target` was not yet
+        reached.
+        """
+        return (
+            self.funds_remaining > 0 and
+            len(self.open_channels) < self.initial_channel_target
+        )
+
+    @property
+    def funds_remaining(self):
+        """The remaining funds after subtracting the already deposited amounts.
+        """
+        if self.funds > 0:
+            remaining = self.funds - sum(
+                channel.deposit for channel in self.open_channels
+            )
+            assert isinstance(remaining, int)
+            return remaining
+        return 0
+
+    @property
+    def open_channels(self):
+        """Shorthand for getting our open channels in this token network.
+        """
```

Comment:
I don't really mind. I just want us to be consistent. If you can switch to closing `"""` at the same line if it's under the character limit that would be great.

I don't think it happens often anyway since by nature docstrings contain a lot of information :)